### DISCUSSION TITLE
[ERROR DRT-0255] Maze Route cannot find path of net net70 in worker of routeBox

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD-flow-scripts#741] [General]
What does this mean and is there anything I can do about it?

Is AutoTuner designed to adjust parameters to find parameters that work around such problems?

E.g. I thought I might try to increase or decresase PLACE_DENSITY to see if that might be the problem.

Can the GUI tell me something? How would I look?

```
[INFO DRT-0195] Start 0th optimization iteration.
    Completing 10% with 113 violations.
    elapsed time = 00:02:16, memory = 58237.49 (MB).
    Completing 20% with 808 violations.
    elapsed time = 00:04:33, memory = 58326.27 (MB).
Failed to find a path between pin  and source aps:
( 228 187 4 ) (Idx) / ( 159716 812722 ) (coords)
extBBox (xDim, yDim, zDim) = (600, 713, 10)
[ERROR DRT-0255] Maze Route cannot find path of net net70 in worker of routeBox ( 158760 812700 ) ( 162540 816480 ).
    Completing 30% with 986 violations.
    elapsed time = 00:06:52, memory = 58426.12 (MB).
Error: detail_route.tcl, 47 DRT-0255
```

Looking at the source code, ```nextPin->getName()``` doesn't have a name(it is an empty string):

https://github.com/The-OpenROAD-Project/OpenROAD/blob/d297c5b18d08684b550e74a3115d4c366bd92e01/src/drt/src/dr/FlexDR_maze.cpp#L2914

**Comment by vijayank88** [comment]
@oharboe 
its good to open a issue with test case.
Also you can check the `net70` in `openroad -gui` where its located or it has enough space to do routing.?

**Reply by oharboe** [reply]
I can't provide the source code in this case and I don't have a small example.

I didn't know how to look at a specific net, but I see "Search" in the GUI can find it for me.

Thanks!

**Reply by vijayank88** [reply]
Explained GUI usage steps here: https://openroad-flow-scripts.readthedocs.io/en/latest/tutorials/FlowTutorial.html#visualizing-design-objects-and-connectivity

**Comment by oharboe** [comment]
Hmm.... is the result of detailed routing deterministic across machine configurations?

I had the failure on one machine, but not another. A difference between the machines is the number of cores.

Also, I see a different number of violations:

```
[INFO DRT-0195] Start 0th optimization iteration.
    Completing 10% with 103 violations.
    elapsed time = 00:02:55, memory = 57651.05 (MB).
    Completing 20% with 825 violations.
    elapsed time = 00:05:56, memory = 56755.36 (MB).
Failed to find a path between pin  and source aps:
( 228 187 4 ) (Idx) / ( 159716 812722 ) (coords)
extBBox (xDim, yDim, zDim) = (600, 713, 10)
[ERROR DRT-0255] Maze Route cannot find path of net net70 in worker of routeBox ( 158760 812700 ) ( 162540 816480 ).
    Completing 30% with 991 violations.
```

Will ```net70``` be the same net on two different machines, if the number of cores differ?

**Reply by oharboe** [reply]
I've studied ```net70``` it is a straight line from a macro to the edge alongside lots of other straight line routes.

This is curious: I haven't seen problems with such straight line routes before. Perhaps a straight line is tricky for a maze router :-)

![image](https://user-images.githubusercontent.com/2798822/210138161-92778fc6-44a9-4726-8f36-e226b0687eed.png)

**Comment by maliberty** [comment]
Its a rather internal error so it will be tricky to debug without a test.  Can you turn on the gcell grid (under Misc) in the GUI image? I'm guessing the net lies on the gcell edge?

The router should be deterministic so the variation across machines is concerning.

**Reply by oharboe** [reply]
Not sure what to make of this. These are the end-points with gridcells enabled. Yes: the net appears to be close to the grid-cell line.

![image](https://user-images.githubusercontent.com/2798822/210152867-532b9248-f558-43ee-b555-7e478f6b8088.png)
![image](https://user-images.githubusercontent.com/2798822/210150714-c233951b-409b-4eb4-801a-96ee3243e77c.png)

**Comment by maliberty** [comment]
I wonder if you could try https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/etc/deltaDebug.py to reduce this to a simple enough test case that you could share it.  You only need to iterate on the drt step.

**Comment by oharboe** [comment]
This was sorted using deltaDebug.py and communicating the confidential case with Matt directly.
